### PR TITLE
Atom: Populate <summary> correctly if it comes after <content>

### LINKED ISSUE #59: feedparser depends on order of description and content
parser  is affected by order by descriotion and content:encoded.
A test case is below.

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:content="http://purl.org/rss/1.0/modules/content/">
<channel>
    <title>title</title>
    <link>http://www.example.com/</link>
    <item>
        <title>title2</title>
        <description>hoge</description>
        <content:encoded><![CDATA[
                fuga
        ]]></content:encoded>
        <link>http://example.com/2.html</link>
    </item>
    <item>
        <title>title1</title>
        <content:encoded><![CDATA[
                fuga
        ]]></content:encoded>
        <description>hoge</description>
        <link>http://example.com/1.html</link>
    </item>
</channel>
</rss>

```

Above two entries' description and content:encoded are just same except order.
But the result is not same..

```
In [4]: a.entries[0].content
Out[4]: [{'base': '', 'language': None, 'type': 'text/html', 'value': 'fuga'}]
In [6]: a.entries[1].content
Out[6]:
[{'base': '', 'language': None, 'type': 'text/html', 'value': 'fuga'},
 {'base': '', 'language': None, 'type': 'text/plain', 'value': 'hoge'}]

In [5]: a.entries[0].description
Out[5]: 'hoge'
In [7]: a.entries[1].description
Out[7]: 'fuga'
```

It seems because 
(1)content is copied to sumary
https://github.com/kurtmckee/feedparser/blob/develop/feedparser/namespaces/_base.py#L482-L483
(2)summary is set to content
https://github.com/kurtmckee/feedparser/blob/develop/feedparser/namespaces/_base.py#L428-L430
this behavior is affected by order, it seems strange to me.

### PR DESCRIPTION
**Pull request by cristoper**
This fixes #59. It is merely a rebase (with some variable renaming) of @mosasiru's original pull request (PR #60) which was never merged for some reason so that it doesn't conflict with the current develop branch.

### CODE REVIEW

File: feedparser/mixin.py [hunk 1]
```diff
@@ -193,6 +193,7 @@ def __init__(self):
         self.svgOK = 0
         self.title_depth = -1
         self.depth = 0
+        self.hasContent = 0
```
Comment (rooterkyberian) [inline feedparser/mixin.py:196]: any reason for it not being:
* bool (True/False)
* snake case?

Comment (cristoper) [reply feedparser/mixin.py:196]: Not that I can see, other than to be consistent with the other state variables (infeed, inentry, etc.) All lowercase would be more consistent (I guess `svgOK` and `title_depth` are the odd ones).

Comment (kurtmckee) [reply feedparser/mixin.py:196]: I'll accept it as-is. There is a _lot_ of legacy code and formatting in feedparser (which is ~17 years old now!), and it'll take some work to get that standardized.

**Review by kurtmckee — APPROVED** [review]

**Comment by kurtmckee** [comment]
This is released as feedparser 6.0.10. Thanks for your work on this!

**Pull request merged**